### PR TITLE
162 add checkstyle and spotbugs into build pipeline

### LINKED ISSUE #162: Add checkstyle and spotbugs into build pipeline

### PR DESCRIPTION
**Pull request by forkimenjeckayang**
Added the Checkstyle and Spotbugs plugin to pom.xml for static code analysis

### CODE REVIEW

**Review by Thendo20 — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -123,6 +123,10 @@
         <maven-gpg-plugin.version>3.1.0</maven-gpg-plugin.version>
         <maven-source-plugin.version>3.0.1</maven-source-plugin.version>
         <nexus-staging-maven-plugin.version>1.6.13</nexus-staging-maven-plugin.version>
+        <checkstyle-maven-plugin.version>3.2.0</checkstyle-maven-plugin.version>
```
Comment (max402) [inline pom.xml]: there is already version [3.4.0](https://mvnrepository.com/artifact/org.apache.maven.plugins/maven-checkstyle-plugin/3.4.0)

File: pom.xml [hunk 2]
```diff
@@ -352,6 +356,68 @@
...
+                            <goal>check</goal>
+                        </goals>
+                        <configuration>
+                            <configLocation>checkstyle.xml</configLocation>
+                            <includeTestSourceDirectory>true</includeTestSourceDirectory>
+                            <encoding>UTF-8</encoding>
+                            <consoleOutput>true</consoleOutput>
+                            <failsOnError>true</failsOnError>
+                            <linkXRef>false</linkXRef>
+                            <outputFile>${project.basedir}/target/checkstyle-report.xml</outputFile>
```
Comment (max402) [inline pom.xml:378]: please add
`<propertyExpansion>org.checkstyle.google.severity=warning</propertyExpansion>`
So we can make it more strict later by changing to `error`

**Review by max402 — APPROVED** [review]
There are a lot of bugs detected by checkstyle and spotbug. We need to create new task to review and fix them

File: pom.xml (hunk 2 above)
Comment (forkimenjeckayang) [reply pom.xml:378]: Great i will correct that, and push before merging

File: pom.xml (hunk 1 above)
Comment (forkimenjeckayang) [reply pom.xml]: i will update the version here too

**Pull request merged**